pkg/reconciler/pipelinerun/affinity_assistant.go

```diff
@@ -68,6 +68,32 @@ func (c *Reconciler) createAffinityAssistants(ctx context.Context, wb []v1beta1.
 				if err == nil {
 					logger.Infof("Created StatefulSet %s in namespace %s", affinityAssistantName, namespace)
 				}
+			case err == nil && a != nil && a.Status.ReadyReplicas == 1:
```

Comment:
This case is just where the statefulset does exist right? what's the reason to check ready replicas?

Reply:
Depending on the infrastructure, it might take a few seconds (I have seen a delay of 20 seconds in some cases) after the time `statefulsets.apps` is created and the required replicas are spawned. This check avoids unnecessary API calls until the affinity assistant pod is created.

Reply:
Would you mind adding a quick comment about this to make it clear why we're checking ready replicas? Also, the existing comment is formatted a bit weirdly; it looks like it's associated with the previous case.

Reply:
Unfortunately, I don't know how to format the comment, it does look weird but if you want that block of comment just before `case`, this is how it shows:

```
			// check whether the affinity assistant (StatefulSet) exists and the affinity assistant pod is created
			// this check requires the StatefulSet to have the available replicas set to 1 to allow any delay between the StatefulSet creation
			// and the necessary pod creation, the delay can be caused by any dependency on PVCs and PVs creation
			// this case addresses issues specified in https://github.com/tektoncd/pipeline/issues/6586
		case err == nil && a != nil && a.Status.ReadyReplicas == 1:
``` 

😞

Reply:
one possibility is:

```
var t interface{}
t = functionOfSomeType()
switch t := t.(type) {
case bool:
    fmt.Printf("boolean %t\n", t)             // t has type bool
``` 

but our comments are much more verbose ...

Reply:
I found a way to indent it such that the comments starts at the same level as `case` 🎉 🤞